### PR TITLE
Document functions of util namespaces - closes #2264

### LINKED ISSUE #2264: Add docstrings to namespaces in util directory
Following up with #2060 we have a ton more namespaces that shed a tear every time someone curses Light Table for not having adequate documentation.

That said, this round focuses on namespaces in the util directory:

- util/
  - cljs.cljs
  - dom.cljs
  - events.cljs
  - ipc.cljs
  - js.cljs
  - kahn.cljs
  - load.cljs
  - style.cljs

### PR DESCRIPTION
**Pull request by sbauer322**
This PR handles the documentation of the files under the util directory:

  - cljs.cljs
  - dom.cljs
  - events.cljs
  - ipc.cljs
  - js.cljs
  - kahn.cljs
  - load.cljs
  - style.cljs

One thing to consider is if we should adjust our project.clj to allow for these namespaces to be generated with Codox. Currently, we have only a small number of namespaces exposed.

See issue #2264

### CODE REVIEW

File: src/lt/util/dom.cljs [hunk 1]
```diff
@@ -131,92 +268,162 @@
...
+  `evs` should be a map of the form `{:ev cb}` where `:ev` is the name and
+  `cb` the callback function."
+  [elem evs]
   (doseq [[ev cb] evs]
     (.addEventListener elem (->ev ev) cb)))
 
-(defn active-element []
+(defn active-element
+  "Return the active element of the document. An element is considered active if
+  it is currently focused."
```
Comment (Mouvedia) [inline src/lt/util/dom.cljs]: > An active element does not necessarily have focus, but an element with focus is always the active element in a document.

File: src/lt/util/js.cljs [hunk 2]
```diff
@@ -1,37 +1,70 @@
...
+(defn wait
+  "Wait `ms` milliseconds before executing `func`."
+  [ms func]
   (js/setTimeout func ms))
 
-(defn now []
+(defn now
+  "Return the current time in milliseconds starting from the Unix epoch."
+  []
   (.getTime (js/Date.)))
```
Comment (Mouvedia) [inline src/lt/util/js.cljs]: If it's supported, this should probably use `Date.now()` instead.

File: src/lt/util/js.cljs [hunk 3]
```diff
@@ -1,37 +1,70 @@
...
-               "'" "&#39;"
-               "/" "&#x2F;"})
+(def entities
+  "Map of entities, such as `&`, to their corresponding character reference."
+  {"&" "&amp;"
+   "<" "&lt;"
+   ">" "&gt;"
+   "\"" "&quot;"
+   "'" "&#39;"
+   "/" "&#x2F;"})
```
Comment (Mouvedia) [inline src/lt/util/js.cljs]: If you ever add $ it should be `&#36;`, not `&dol;` or `&dollar;`.

File: src/lt/util/dom.cljs [hunk 4]
```diff
@@ -37,92 +38,228 @@
...
      (or (.item this n) not-found))))
 
-(defn text-node [text]
+(defn text-node
+  "Create a text node containing string `text`."
+  [text]
   (js/document.createTextNode text))
 
 (defn $$
+  "Returns a NodeList of all elements within `elem` that matches `query`.
```
Comment (sbauer322) [inline src/lt/util/dom.cljs]: To clarify, did you mean "+that *match"?

"within `elem` the match query" does not seem correct.

File: src/lt/util/dom.cljs [hunk 5]
```diff
@@ -37,92 +38,228 @@
...
+  "Remove `elem` from the DOM tree."
+  [elem]
   (when-let [p (parent elem)]
     (.removeChild p elem)))
 
-(defn empty [elem]
+(defn empty
+  "Sets the inner HTML of `elem` to an empty string."
+  [elem]
   (set! (.-innerHTML elem) ""))
```
Comment (sbauer322) [inline src/lt/util/dom.cljs]: Are you asking with regards to the docstring, or about the actual code?

File: src/lt/util/js.cljs (hunk 3 above)
Comment (sbauer322) [inline src/lt/util/js.cljs]: Thanks for the tip, I've updated the things to [refactor wiki page](https://github.com/LightTable/LightTable/wiki/Possible-Refactoring-Targets) with this info.

File: src/lt/util/js.cljs (hunk 2 above)
Comment (sbauer322) [inline src/lt/util/js.cljs]: Thanks for the tip, I've updated the things to [refactor wiki page](https://github.com/LightTable/LightTable/wiki/Possible-Refactoring-Targets) with this info.

File: src/lt/util/dom.cljs (hunk 5 above)
Comment (sbauer322) [inline src/lt/util/dom.cljs]: I suspect you meant the code, in which case, I've updated the [refactor wiki page](https://github.com/LightTable/LightTable/wiki/Possible-Refactoring-Targets) with this info.

File: src/lt/util/dom.cljs [hunk 6]
```diff
@@ -131,92 +268,162 @@
...
-(defn replace-with [orig neue]
+(defn replace-with
+  "Replace `orig` with `neue`."
+  [orig neue]
   (when-let [p (parent orig)]
     (.replaceChild p neue orig)))
 
-(defn height [elem]
+(defn height
+  "Return the height of `elem`, as an integer."
```
Comment (sbauer322) [inline src/lt/util/dom.cljs]: This docstring has been applied to the PR.

File: src/lt/util/dom.cljs (hunk 1 above)
Comment (sbauer322) [inline src/lt/util/dom.cljs]: This docstring has been applied to the PR.

File: src/lt/util/dom.cljs (hunk 4 above)
Comment (Mouvedia) [inline src/lt/util/dom.cljs]: by "+the" I meant "all _the_ elements"

You shouldn't take this review seriously anyway: I just wanted to try that new Github feature.

File: src/lt/util/dom.cljs [hunk 7]
```diff
@@ -37,92 +38,228 @@
...
+  (dom/set-attr ($ \"div .tabsets\") {:draggable \"true\"})
+  ;;=> nil
+  ;; div is now: <div class=\"tabsets\" draggable=\"true\";>
+  ```"
+  [elem things]
   (doseq [[k v] things]
     (.setAttribute elem (name k) (if (keyword? v) (name v) v))))
 
-(defn attr [elem things]
+(defn attr
```
Comment (rundis) [inline src/lt/util/dom.cljs]: kittens are dying. It pains me to see this. Awful api design. Needs to be deprecated at some point in the future. Ah well, nothing to do with your PR, but had do get it out.

**Comment by rundis** [comment]
I've made a few comments. Nothing major. Have a look, fix what you think is worth addressing. Then feel free to merge this. Tx for doing this !

File: src/lt/util/dom.cljs (hunk 7 above)
Comment (Mouvedia) [inline src/lt/util/dom.cljs]: @rundis #2273

Comment (sbauer322) [inline src/lt/util/dom.cljs]: As I didn't want to futher add to the issue pile, I created this [refactoring targets wiki](https://github.com/LightTable/LightTable/wiki/Possible-Refactoring-Targets) page for other changes.

**Comment by sbauer322** [comment]
Maybe I'm missing them, @rundis, but I do not see any comments beyond the awful api design remark. Were there any others you had in mind to address prior to merging?

**Comment by rundis** [comment]
Some of My comments was still pending. Sorry.

File: src/lt/util/load.cljs [hunk 8]
```diff
@@ -1,23 +1,44 @@
...
+  (absolute? \"./foo/bar\")        ;;=> false
+
+  (absolute? \"foo/bar\")          ;;=> false
+  ```"
+  [path]
   (boolean (re-seq #"^\s*[\\\/]|([\w]+:[\\\/])" path)))
 
-(defn node-module [path]
+(defn node-module
+  "Requires the node module located at `path`."
```
Comment (sbauer322) [inline src/lt/util/load.cljs]: I believe so. This is what I am going with, anyways!

File: src/lt/util/load.cljs [hunk 9]
```diff
@@ -1,23 +1,44 @@
 (ns lt.util.load
-  "Provide fns to load js, css and node module assets into LT"
+  "Provide functions to load js, css and node module assets into LT."
   (:require [clojure.string :as string]))
 
-(def fpath (js/require "path"))
-(def fs (js/require "fs"))
+(def fpath "Provides access to fpath library." (js/require "path"))
+(def fs "Provides access to fs library." (js/require "fs"))
```
Comment (sbauer322) [inline src/lt/util/load.cljs]: Yes, the clarification and link would be useful.

**Pull request merged**